src/wit_gen.rs

```diff
@@ -124,20 +148,61 @@ impl<F: Field + Hash> FixedGenContext<F> {
             panic!("trying to assign non-fixed signal");
         }
 
-        if let Some(assigments) = self.assigments.get_mut(&lhs) {
-            assigments[offset] = rhs;
+        if let Some(assignments) = self.assignments.get_mut(&lhs) {
+            assignments[offset] = rhs;
         } else {
-            let mut assigments = vec![F::ZERO; self.num_steps];
-            assigments[offset] = rhs;
-            self.assigments.insert(lhs, assigments);
+            let mut assignments = vec![F::ZERO; self.num_steps];
+            assignments[offset] = rhs;
+            self.assignments.insert(lhs, assignments);
         }
     }
 
-    pub fn get_assigments(self) -> FixedAssignment<F> {
-        self.assigments
+    pub fn get_assignments(self) -> FixedAssignment<F> {
+        self.assignments
     }
 
     fn is_fixed_queriable(q: Queriable<F>) -> bool {
         matches!(q, Queriable::Halo2FixedQuery(_, _) | Queriable::Fixed(_, _))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{dsl::StepTypeWGHandler, util::uuid};
+
+    fn dummy_args_fn() {}
+
+    #[test]
+    fn test_padding_no_witness() {
+        let mut ctx = TraceContext::new(5);
+        let step = StepTypeWGHandler::new(
+            uuid(),
+            "dummy",
+            |_: &mut StepInstance<i32>, _: ()| {}, // replace i32 with your Field type
```

Comment:
I don't think the comment is useful. Nobody should come and replace that in this test